homeassistant/components/synology_dsm/__init__.py

```diff
@@ -114,111 +87,50 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             entry, data={**entry.data, CONF_MAC: network.macs}
         )
 
-    async def async_coordinator_update_data_cameras() -> dict[
-        str, dict[str, SynoCamera]
-    ] | None:
-        """Fetch all camera data from api."""
-        if not hass.data[DOMAIN][entry.unique_id][SYSTEM_LOADED]:
-            raise UpdateFailed("System not fully loaded")
-
-        if SynoSurveillanceStation.CAMERA_API_KEY not in api.dsm.apis:
-            return None
-
-        surveillance_station = api.surveillance_station
-        current_data: dict[str, SynoCamera] = {
-            camera.id: camera for camera in surveillance_station.get_all_cameras()
-        }
-
-        try:
-            async with async_timeout.timeout(30):
-                await hass.async_add_executor_job(surveillance_station.update)
-        except SynologyDSMAPIErrorException as err:
-            raise UpdateFailed(f"Error communicating with API: {err}") from err
-
-        new_data: dict[str, SynoCamera] = {
-            camera.id: camera for camera in surveillance_station.get_all_cameras()
-        }
-
-        for cam_id, cam_data_new in new_data.items():
-            if (
-                (cam_data_current := current_data.get(cam_id)) is not None
-                and cam_data_current.live_view.rtsp != cam_data_new.live_view.rtsp
-            ):
-                async_dispatcher_send(
-                    hass,
-                    f"{SIGNAL_CAMERA_SOURCE_CHANGED}_{entry.entry_id}_{cam_id}",
-                    cam_data_new.live_view.rtsp,
-                )
-
-        return {"cameras": new_data}
-
-    async def async_coordinator_update_data_central() -> None:
-        """Fetch all device and sensor data from api."""
+    # These all create executor jobs so we do not gather here
+    coordinator_central = SynologyDSMCentralUpdateCoordinator(hass, entry, api)
+    await coordinator_central.async_config_entry_first_refresh()
```

Comment:
I got rid of the SYSTEM_LOADED code that was added in #46113 by refreshing here before we setup the other coordinators.   I'm not sure if thats a problem but it works fine on my machines

Reply:
The SYSTEM_LOADED code was added to avoid predictable API call errors during service call or button triggered reboot triggered